File: 2017SD - University final project (web)/Code/src/main/java/acu/project1/controller/CourseController.java
package acu.project1.controller;

import org.jsondoc.core.annotation.Api;
import org.jsondoc.core.annotation.ApiMethod;
import org.jsondoc.core.annotation.ApiPathParam;
import org.jsondoc.core.pojo.ApiStage;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import acu.project1.business.service.CourseService;
import acu.project1.business.transfer.CourseData;
import acu.project1.business.transfer.StudentData;

import java.util.List;

@RestController
@RequestMapping(value = "/course")
@Api(
        name = "Course API",
        description = "Provides a list of courses",
        stage = ApiStage.RC)
public class CourseController {

    private CourseService courseService;
   
    @Autowired
    public CourseController(CourseService courseService){
        this.courseService = courseService;
    }

    @RequestMapping(value = "/all", method = RequestMethod.GET)
    @ApiMethod(description = "Get all courses from the database")
    public List<CourseData> getAll(){
        return courseService.getAllCourses();
    }
    
    @RequestMapping(value = "/get/{id}", method = RequestMethod.GET)
    @ApiMethod(description = "Get all students from the course")
    public List<StudentData> getAllStudentsOfCourse(@ApiPathParam(name = "id") @PathVariable long id){
        return courseService.getAllCourseStudents(id);
    }
    
    @RequestMapping(value = "/enroll/{id1}/{id2}", method = RequestMethod.GET)
    @ApiMethod(description = "Get all students from the course")
    public List<StudentData> getAllStudentsOfCourse(@ApiPathParam(name = "id1") @PathVariable long id1, @ApiPathParam(name = "id2") @PathVariable long id2){
        courseService.addStudentToCourse(id2, id1);
        return courseService.getAllCourseStudents(id1);
    }
    
    @RequestMapping(value = "/create/{name}/", method = RequestMethod.POST)
    @ApiMethod(description = "Create a course and save it to the database")
    public List<CourseData> create(@ApiPathParam(name = "name") @PathVariable String name){
        courseService.addCourse(name);
        return courseService.getAllCourses();
    }

    @RequestMapping(value = "/delete/{id}", method = RequestMethod.POST)
    @ApiMethod(description = "Remove the course with the provided ID from the database")
    public List<CourseData> remove(@ApiPathParam(name = "id") @PathVariable long id){
        courseService.removeCourse(id);

        return courseService.getAllCourses();
    }
}

File: temp/src/main/resources/static/app/course.controller.js
(function () {
    'use strict';

    angular
        .module('app')
        .controller('CourseController', CourseController);

    CourseController.$inject = ['$http'];

    function CourseController($http) {
        var vm = this;

        vm.returnList = [];
        vm.getAll = getAll;
        vm.students = [];
        vm.enroll = enroll;
        vm.getStudents = getStudents;
        vm.createCourse = createCourse;
        vm.deleteCourse = deleteCourse;
        init();

        function init(){
            getAll();
            getStudents(1);
        }

        function getAll(){
            var url = "/course/all";
            var coursesPromise = $http.get(url);
            coursesPromise.then(function(response){
                vm.returnList = response.data;
            });
        }

        function getStudents(id){
        	 var url = "/course/get/" + id;
             var coursesPromise = $http.get(url);
             coursesPromise.then(function(response){
                 vm.students = response.data;
             });
        }
        function createCourse(name){
        	var url = "/course/create/" + name;
        	$http.post(url).then(function(response){
        		vm.returnList = response.data;
        	});
        }
        function enroll(id1,id2){
        	//getAll();
        	var url = "/course/enroll/" + id1 + "/" + id2;
        	var coursesPromise = $http.get(url);
            coursesPromise.then(function(response){
                vm.students = response.data;
            });
            //getStudents(1);
        }
        
        function deleteCourse(id){
            var url = "/course/delete/" + id;
            $http.post(url).then(function(response){
                vm.returnList = response.data;
            });
        }
    }
})();
